### ISSUE TITLE
AnimatedContent is now stable

### ISSUE THREAD

**Issue by yurii-shcherbiuk** [adrielcafe/voyager#120]
https://developer.android.com/jetpack/androidx/releases/compose-animation#1.5.0-alpha01

![Error](https://user-images.githubusercontent.com/93974439/227710559-136b6fa8-917e-4502-a533-09d3eccef6a2.jpg)

**Comment by DevSrSouza** [comment]
Seen this version is still Alpha, we will wait until get Stable to update the Animations module and not break people that is using Stable versions of Compose.

**Comment by DevSrSouza** [comment]
Recommendation for now: Copy voyager code to your project with Compose Alpha version and do the proper dependency changes.

We this update get Stable on both Compose Android and Compose Multiplatform (Jetbrains fork) we will update it and publish a new version!

**Comment by CodeQuacker** [comment]
for now to make transistions work:

1.  Copy ScreenTransistion and specific transistion you want to use (for example SlideTransistion)
2. Change AnimatedContentScope<Screen> to AnimatedContentTransitionScope<Screen> in both ScreenTransistion functions
3. Make sure your specific transistion uses your new ScreenTransistion (imports)

My end result:

- ScreenTransistion:
```kotlin
package <package name>

import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.AnimatedContentTransitionScope
import androidx.compose.animation.AnimatedVisibilityScope
import androidx.compose.animation.ContentTransform
import androidx.compose.animation.ExperimentalAnimationApi
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import cafe.adriel.voyager.core.screen.Screen
import cafe.adriel.voyager.core.stack.StackEvent
import cafe.adriel.voyager.navigator.Navigator

public typealias ScreenTransitionContent = @Composable AnimatedVisibilityScope.(Screen) -> Unit

@ExperimentalAnimationApi
@Composable
public fun ScreenTransition(
    navigator: Navigator,
    enterTransition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
    exitTransition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
    modifier: Modifier = Modifier,
    content: ScreenTransitionContent = { it.Content() }
) {
    ScreenTransition(
        navigator = navigator,
        modifier = modifier,
        content = content,
        transition = {
            when (navigator.lastEvent) {
                StackEvent.Pop -> exitTransition()
                else -> enterTransition()
            }
        }
    )
}

@ExperimentalAnimationApi
@Composable
public fun ScreenTransition(
    navigator: Navigator,
    transition: AnimatedContentTransitionScope<Screen>.() -> ContentTransform,
    modifier: Modifier = Modifier,
    content: ScreenTransitionContent = { it.Content() }
) {
    AnimatedContent(
        targetState = navigator.lastItem,
        transitionSpec = transition,
        modifier = modifier,
        label = ""
    ) { screen ->
        navigator.saveableState("transition", screen) {
            content(screen)
        }
    }
}
``` 
- SlideTransistion:
```kotlin
package <package name>

import androidx.compose.animation.ExperimentalAnimationApi
import androidx.compose.animation.core.FiniteAnimationSpec
import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.VisibilityThreshold
import androidx.compose.animation.core.spring
import androidx.compose.animation.slideInHorizontally
import androidx.compose.animation.slideInVertically
import androidx.compose.animation.slideOutHorizontally
import androidx.compose.animation.slideOutVertically
import androidx.compose.animation.with
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.IntOffset
import cafe.adriel.voyager.core.stack.StackEvent
import cafe.adriel.voyager.navigator.Navigator

@ExperimentalAnimationApi
@Composable
public fun SlideTransition(
    navigator: Navigator,
    modifier: Modifier = Modifier,
    orientation: SlideOrientation = SlideOrientation.Horizontal,
    animationSpec: FiniteAnimationSpec<IntOffset> = spring(
        stiffness = Spring.StiffnessMediumLow,
        visibilityThreshold = IntOffset.VisibilityThreshold
    ),
    content: ScreenTransitionContent = { it.Content() }
) {
    ScreenTransition(
        navigator = navigator,
        modifier = modifier,
        content = content,
        transition = {
            val (initialOffset, targetOffset) = when (navigator.lastEvent) {
                StackEvent.Pop -> ({ size: Int -> -size }) to ({ size: Int -> size })
                else -> ({ size: Int -> size }) to ({ size: Int -> -size })
            }

            when (orientation) {
                SlideOrientation.Horizontal ->
                    slideInHorizontally(animationSpec, initialOffset) with
                            slideOutHorizontally(animationSpec, targetOffset)

                SlideOrientation.Vertical ->
                    slideInVertically(animationSpec, initialOffset) with
                            slideOutVertically(animationSpec, targetOffset)
            }
        }
    )
}

public enum class SlideOrientation {
    Horizontal,
    Vertical
}
```

**Comment by akardas16** [comment]
android studio shows depreciation on `with ` and sugests to replace `togetherWith`

**Comment by adrielcafe** [comment]
Fixed on [1.0.0-rc07](https://github.com/adrielcafe/voyager/releases/tag/1.0.0-rc07).
